enterprise/app/org/org_access_denied.tsx

```diff
@@ -0,0 +1,42 @@
+import { User } from "../../../app/auth/user";
+import React from "react";
+import FilledButton from "../../../app/components/button/button";
+import authService from "../../../app/auth/auth_service";
+import router from "../../../app/router/router";
+
+export type Props = {
+  user: User;
+};
+
+export default class OrgAccessDeniedComponent extends React.Component<Props> {
+  handleImpersonateClicked() {
+    const params = new URLSearchParams(window.location.search);
+    const sourceUrl = params.get("source_url");
+    if (sourceUrl) {
+      router.navigateTo(sourceUrl);
+    }
+    authService.enterImpersonationMode(this.props.user.subdomainGroupID);
+  }
+
+  render() {
+    return (
+      <div className="state-page">
+        <div className="shelf">
+          <div className="container">
+            <div className="titles">
+              <div className="title">Access denied</div>
+            </div>
+            <div className="details">You are not authorized to access this site.</div>
+            {this.props.user?.subdomainGroupID && (
+              <div>
+                <FilledButton onClick={this.handleImpersonateClicked.bind(this)} className="impersonate-button">
+                  Impersonate owner
+                </FilledButton>
+              </div>
```

Comment:
probably want to hide this behind `this.props.user?.canCall('getInvocationOwner')` so we don't show the button if the user can't impersonate. (I know the RPC-based perms feel a little clunky so if you want you could also add a method like `canImpersonate()` to the User class which just calls that under the hood for now)

Reply:
subdomainGroupID is not populated if the user is not allowed to impersonate